haskell-tests/Types/CBOR.hs

```diff
@@ -6,35 +6,47 @@ import Codec.CBOR.Read
 import qualified Codec.CBOR.Term as CBOR
 import Codec.CBOR.Write
 import qualified Codec.CBOR.Write as CBOR
+import qualified Concordium.Crypto.SHA256 as SHA256
+import Concordium.Types
+import Concordium.Types.ProtocolLevelTokens.CBOR
+import Concordium.Types.Queries.Tokens
+import Control.Monad
 import qualified Data.Aeson as AE
 import qualified Data.Aeson.KeyMap as AE
 import qualified Data.ByteString as BS
 import qualified Data.ByteString.Short as BSS
+import qualified Data.FixedByteString as FBS
 import qualified Data.Map as Map
 import qualified Data.Sequence as Seq
 import qualified Data.Text as Text
 import qualified Data.Text.Encoding as Text
 import qualified Data.Vector as V
+import Generators
 import Test.HUnit
 import Test.Hspec
 import Test.QuickCheck
 
-import Concordium.Types
-import Concordium.Types.ProtocolLevelTokens.CBOR
-import Concordium.Types.Queries.Tokens
-import qualified Data.FixedByteString as FBS
-import Generators
-
 genText :: Gen Text.Text
 genText = sized $ \s -> Text.decodeUtf8 . BS.pack <$> genUtf8String s
 
 genTokenAmount :: Gen TokenAmount
 genTokenAmount = TokenAmount <$> arbitrary <*> chooseBoundedIntegral (0, 255)
 
+genSha256Hash :: Gen SHA256.Hash
+genSha256Hash = do
+    randomBytes <- BS.pack <$> replicateM SHA256.digestSize (choose (0, 255)) -- Generate 32 random bytes
+    return (SHA256.hash randomBytes)
+
+genTokenMetadataUrl :: Gen TokenMetadataUrl
+genTokenMetadataUrl = do
+    url <- genText
+    checksumSha256 <- Just <$> genSha256Hash
+    return TokenMetadataUrl{tmUrl = url, tmChecksumSha256 = checksumSha256, tmAdditional = Map.empty}
```

Comment:
I think we also need tests for the case where `tmAdditional` is non-empty (cf. `genTGokenModuleStateWithAdditional`).